src/Entity/Sylius/OrderRepository.php

```diff
@@ -245,4 +245,21 @@ public static function addVendorClause(QueryBuilder $qb, $alias, LocalBusiness $
             ->andWhere(sprintf('%s.restaurant = :restaurant', $vendorAlias))
             ->setParameter('restaurant', $restaurant);
     }
+
+    public function fetchNextSeqId(){
+        $dbConnection = $this->getEntityManager()->getConnection();
+        $nextValQuery = $dbConnection->getDatabasePlatform()->getSequenceNextValSQL('sylius_order_id_seq');
```

Comment:
Is the sequence value updated in the table? I'm not entirely sure if I've grasped it correctly, but from what I gather, there seems to be a cycle of events:

For instance:
1. If Order `#100` encounters a collision with an existing number,
2. You fetch the next ID using `getSequenceNextValSQL()` and get `101`.
3. Subsequently, you store `Crockford::encode(101)` as the number for Order `#100`.
4. However, when Order `#101` is generated, won't its number collide with Order `#101` and so forth?

Reply:
Sorry, I didn't include information about how this works (info that I didn't know either until I need to implement this by the way).
The `nextval` function in PostgreSQL advances the sequence to its next value and returns that value. So in that example would be:

1. Order #100 encounters a collision with an existing number
2. Fetch next ID available (`101`) and advance the sequence
3. Store Crockford::encode(101) as the number for Order #100.
4. Next Order will be generated with the ID 102, the Order with 101 won't exist in the platform since we advance the sequence manually and use that ID to resolve the collision.

Reply:
Okay perfect then ! That was my question / concern :+1: